TypeScript error: intersectsTriangle not recognized on boundsTree.shapecast()

### Description

After upgrading from 0.9.4 to 0.9.7, TypeScript rejects intersectsTriangle callback in shapecast() calls on geometry.boundsTree.                      
                                                                                                                                                        
The root cause: boundsTree is typed as GeometryBVH, but computeBoundsTree() returns MeshBVH by default. Since intersectsTriangle is only defined on MeshBVH.shapecast() (not the base ShapecastCallbacks interface), TypeScript errors:
```                                                                                                                                                       
error TS2561: Object literal may only specify known properties, but 'intersectsTriangle' does not exist in type 'ShapecastCallbacks'.
```

The code works at runtime since boundsTree is actually a MeshBVH instance.                     

Introduced in: commit https://github.com/gkjohnson/three-mesh-bvh/commit/ade609eb55e195bc252ac814c9c4c1d4dc51271b ("Update shapecast dts", Jan 8 2026)

### Reproduction steps

1. Create a mesh and compute bounds tree
2. Call shapecast with intersectsTriangle
3. TypeScript reports error on intersectsTriangle


### Code

```ts
const geometry = new BoxGeometry();                                                                                                                   geometry.computeBoundsTree(); 
geometry.boundsTree!.shapecast({
    intersectsBounds: (box) => box.intersectsBox(targetBox)
    intersectsTriangle: (tri) => {
      // TypeScript error here
      return false;
    },
  });
```


### Live example

hard to setup type bugs like this online

### Screenshots

N/A - this is a type-only error, no visual/runtime issue

### Library Version

0.9.7

### Three.js Version

0.182.0

--- a/src/index.d.ts
+++ b/src/index.d.ts
@@ -69,6 +69,14 @@ export interface ShapecastCallbacks {
 		nodeIndex: number,
 		box: Box3
 	) => boolean;
+
+	/** @deprecated geometry.boundsTree must be cast to "MeshBVH" in order to use "intersectsTriangle" callback. */
+	intersectsTriangle?: (
+		triangle: ExtendedTriangle,
+		triangleIndex: number,
+		contained: boolean,
+		depth: number
+	) => boolean|void
 }
 
 export interface BVHCastCallbacks {

